### PR TITLE
vimPlugins: update on 2024-08-05

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

- **vimPlugins: update on 2024-08-05**
- **vimPlugins: resolve github repository redirects**
- **vimPlugins.nvim-treesitter: update grammars**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 332408` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>198 packages built:</summary>
  <ul>
    <li>vimPlugins.ChatGPT-nvim</li>
    <li>vimPlugins.CopilotChat-nvim</li>
    <li>vimPlugins.Coqtail</li>
    <li>vimPlugins.LeaderF</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.SpaceVim</li>
    <li>vimPlugins.Vundle-vim</li>
    <li>vimPlugins.advanced-git-search-nvim</li>
    <li>vimPlugins.alpha-nvim</li>
    <li>vimPlugins.auto-session</li>
    <li>vimPlugins.bamboo-nvim</li>
    <li>vimPlugins.barbecue-nvim</li>
    <li>vimPlugins.base46</li>
    <li>vimPlugins.better-escape-nvim</li>
    <li>vimPlugins.bullets-vim</li>
    <li>vimPlugins.catppuccin-nvim</li>
    <li>vimPlugins.cmp-ai</li>
    <li>vimPlugins.cmp-clippy</li>
    <li>vimPlugins.cmp-copilot</li>
    <li>vimPlugins.cmp-dap</li>
    <li>vimPlugins.cmp-dictionary</li>
    <li>vimPlugins.cmp-digraphs</li>
    <li>vimPlugins.cmp-fish</li>
    <li>vimPlugins.cmp-fuzzy-buffer</li>
    <li>vimPlugins.cmp-fuzzy-path</li>
    <li>vimPlugins.cmp-git</li>
    <li>vimPlugins.cmp-greek</li>
    <li>vimPlugins.cmp-look</li>
    <li>vimPlugins.cmp-neosnippet</li>
    <li>vimPlugins.cmp-npm</li>
    <li>vimPlugins.cmp-nvim-lsp-signature-help</li>
    <li>vimPlugins.cmp-nvim-tags</li>
    <li>vimPlugins.cmp-nvim-ultisnips</li>
    <li>vimPlugins.cmp-pandoc-nvim</li>
    <li>vimPlugins.cmp-rg</li>
    <li>vimPlugins.cmp-snippy</li>
    <li>vimPlugins.cmp-tabby</li>
    <li>vimPlugins.cmp-tmux</li>
    <li>vimPlugins.cmp-vim-lsp</li>
    <li>vimPlugins.cmp-vimwiki-tags</li>
    <li>vimPlugins.cmp-zsh</li>
    <li>vimPlugins.cobalt2-nvim</li>
    <li>vimPlugins.coc-nvim</li>
    <li>vimPlugins.codeium-nvim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.crates-nvim</li>
    <li>vimPlugins.cyberdream-nvim</li>
    <li>vimPlugins.debugprint-nvim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.deol-nvim</li>
    <li>vimPlugins.efmls-configs-nvim</li>
    <li>vimPlugins.eyeliner-nvim</li>
    <li>vimPlugins.firenvim</li>
    <li>vimPlugins.flit-nvim</li>
    <li>vimPlugins.fuzzy-nvim</li>
    <li>vimPlugins.fzf-hoogle-vim</li>
    <li>vimPlugins.fzf-lua</li>
    <li>vimPlugins.fzf-vim</li>
    <li>vimPlugins.git-blame-nvim</li>
    <li>vimPlugins.gitsigns-nvim</li>
    <li>vimPlugins.gleam-vim</li>
    <li>vimPlugins.go-nvim</li>
    <li>vimPlugins.grug-far-nvim</li>
    <li>vimPlugins.gruvbox-material-nvim</li>
    <li>vimPlugins.guard-nvim</li>
    <li>vimPlugins.hardhat-nvim</li>
    <li>vimPlugins.haskell-scope-highlighting-nvim</li>
    <li>vimPlugins.haskell-snippets-nvim</li>
    <li>vimPlugins.idris2-nvim</li>
    <li>vimPlugins.image-nvim</li>
    <li>vimPlugins.lazy-lsp-nvim</li>
    <li>vimPlugins.lean-nvim</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.leap-nvim</li>
    <li>vimPlugins.legendary-nvim</li>
    <li>vimPlugins.lsp-format-nvim</li>
    <li>vimPlugins.lspsaga-nvim</li>
    <li>vimPlugins.luasnip</li>
    <li>vimPlugins.markview-nvim</li>
    <li>vimPlugins.mason-lspconfig-nvim</li>
    <li>vimPlugins.material-vim</li>
    <li>vimPlugins.mini-nvim</li>
    <li>vimPlugins.mkdnflow-nvim</li>
    <li>vimPlugins.multicursors-nvim</li>
    <li>vimPlugins.neoconf-nvim</li>
    <li>vimPlugins.neoformat</li>
    <li>vimPlugins.neogen</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neorg-telescope</li>
    <li>vimPlugins.neoscroll-nvim</li>
    <li>vimPlugins.neotest-golang</li>
    <li>vimPlugins.neotest-haskell</li>
    <li>vimPlugins.neotest-playwright</li>
    <li>vimPlugins.neotest-zig</li>
    <li>vimPlugins.nlsp-settings-nvim</li>
    <li>vimPlugins.none-ls-nvim</li>
    <li>vimPlugins.nvchad</li>
    <li>vimPlugins.nvchad-ui</li>
    <li>vimPlugins.nvim-cmp</li>
    <li>vimPlugins.nvim-dap</li>
    <li>vimPlugins.nvim-dap-python</li>
    <li>vimPlugins.nvim-dap-ui</li>
    <li>vimPlugins.nvim-highlight-colors</li>
    <li>vimPlugins.nvim-jdtls</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-navbuddy</li>
    <li>vimPlugins.nvim-navic</li>
    <li>vimPlugins.nvim-spectre</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-parsers.cuda</li>
    <li>vimPlugins.nvim-treesitter-parsers.d</li>
    <li>vimPlugins.nvim-treesitter-parsers.gomod</li>
    <li>vimPlugins.nvim-treesitter-parsers.groovy</li>
    <li>vimPlugins.nvim-treesitter-parsers.hlsl</li>
    <li>vimPlugins.nvim-treesitter-parsers.http</li>
    <li>vimPlugins.nvim-treesitter-parsers.hurl</li>
    <li>vimPlugins.nvim-treesitter-parsers.just</li>
    <li>vimPlugins.nvim-treesitter-parsers.kotlin</li>
    <li>vimPlugins.nvim-treesitter-parsers.latex</li>
    <li>vimPlugins.nvim-treesitter-parsers.matlab</li>
    <li>vimPlugins.nvim-treesitter-parsers.nix</li>
    <li>vimPlugins.nvim-treesitter-parsers.perl</li>
    <li>vimPlugins.nvim-treesitter-parsers.powershell</li>
    <li>vimPlugins.nvim-treesitter-parsers.problog</li>
    <li>vimPlugins.nvim-treesitter-parsers.prolog</li>
    <li>vimPlugins.nvim-treesitter-parsers.rescript</li>
    <li>vimPlugins.nvim-treesitter-parsers.roc</li>
    <li>vimPlugins.nvim-treesitter-parsers.slang</li>
    <li>vimPlugins.nvim-treesitter-parsers.sql</li>
    <li>vimPlugins.nvim-treesitter-parsers.swift</li>
    <li>vimPlugins.nvim-treesitter-parsers.typespec</li>
    <li>vimPlugins.nvim-treesitter-parsers.vim</li>
    <li>vimPlugins.nvim-treesitter-parsers.vrl</li>
    <li>vimPlugins.nvim-treesitter-parsers.wit</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.nvim-ufo</li>
    <li>vimPlugins.nvim-web-devicons</li>
    <li>vimPlugins.obsidian-nvim</li>
    <li>vimPlugins.octo-nvim</li>
    <li>vimPlugins.oil-nvim</li>
    <li>vimPlugins.onedarkpro-nvim</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.otter-nvim</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.precognition-nvim</li>
    <li>vimPlugins.promise-async</li>
    <li>vimPlugins.rainbow-delimiters-nvim</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.render-markdown</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.roslyn-nvim</li>
    <li>vimPlugins.scope-nvim</li>
    <li>vimPlugins.sideways-vim</li>
    <li>vimPlugins.smart-splits-nvim</li>
    <li>vimPlugins.splitjoin-vim</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.statuscol-nvim</li>
    <li>vimPlugins.tabby-nvim</li>
    <li>vimPlugins.tagbar</li>
    <li>vimPlugins.telescope-cheat-nvim</li>
    <li>vimPlugins.telescope-coc-nvim</li>
    <li>vimPlugins.telescope-file-browser-nvim</li>
    <li>vimPlugins.telescope-frecency-nvim</li>
    <li>vimPlugins.telescope-fzf-native-nvim</li>
    <li>vimPlugins.telescope-fzf-writer-nvim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-live-grep-args-nvim</li>
    <li>vimPlugins.telescope-manix</li>
    <li>vimPlugins.telescope-media-files-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.telescope-symbols-nvim</li>
    <li>vimPlugins.telescope-undo-nvim</li>
    <li>vimPlugins.telescope-z-nvim</li>
    <li>vimPlugins.telescope-zoxide</li>
    <li>vimPlugins.terminus</li>
    <li>vimPlugins.text-case-nvim</li>
    <li>vimPlugins.tmux-nvim</li>
    <li>vimPlugins.typescript-tools-nvim</li>
    <li>vimPlugins.ultimate-autopair-nvim</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vim-dadbod-completion</li>
    <li>vimPlugins.vim-flog</li>
    <li>vimPlugins.vim-fzf-coauthorship</li>
    <li>vimPlugins.vim-graphql</li>
    <li>vimPlugins.vim-just</li>
    <li>vimPlugins.vim-lsp</li>
    <li>vimPlugins.vim-pluto</li>
    <li>vimPlugins.vim-snipmate</li>
    <li>vimPlugins.vim-spirv</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vim-zettel</li>
    <li>vimPlugins.vimspector</li>
    <li>vimPlugins.vimux</li>
    <li>vimPlugins.vscode-nvim</li>
    <li>vimPlugins.wiki-vim</li>
    <li>vimPlugins.yazi-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Note: the update script is indeed significantly faster now !

**Comment by teto** [comment]
> Note: the update script is indeed significantly faster now !

thanks for the heads up ! nvim-treesitter tests fail

**Comment by GaetanLepage** [comment]
> nvim-treesitter tests fail

Yes, but this is not a regression. It has been the case for a while...
https://github.com/NixOS/nixpkgs/pull/323990#issuecomment-2206673630

**Comment by GaetanLepage** [comment]
> I have the memory of a goldfish so I am bound to ask the same question on the next update. Is there a ticket to track that treesitter test error/do we know why it fails ?

No, we should open one indeed. Do you mind going for it ?

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/generated.nix [hunk 1]
```diff
@@ -2555,12 +2555,12 @@
...
     language = "sql";
-    version = "0.0.0+rev=a966446";
+    version = "0.0.0+rev=b817500";
     src = fetchFromGitHub {
       owner = "derekstride";
       repo = "tree-sitter-sql";
-      rev = "a9664463580473e92d8f5e29fa06fb1be88752af";
-      hash = "sha256-0SY6dOofB+zv4xa7oXabEoUZd5NUV1NHhB+Jx6m137I=";
+      rev = "b8175006d9c8120d41cf40a4ef3711bbbbc08973";
+      hash = "sha256-idQB8Wqw7lvU192y7+UgFvcwlmY71/mu9jJ4hRc4ud4=";
```
Comment (kalekseev) [inline pkgs/applications/editors/vim/plugins/nvim-treesitter/generated.nix:2563]: @GaetanLepage I get hash mismatch here 
```
       specified: sha256-idQB8Wqw7lvU192y7+UgFvcwlmY71/mu9jJ4hRc4ud4=
            got:    sha256-ZURUEd7TokBIVVejRoBiXMQ1XwUEgMnzOhQiV+Tdpk0=
error: 1 dependencies of derivation '/nix/store/gkj82kn7lv4npd2y312dxf47q6l07mb7-sql-grammar-0.0.0+rev=b817500.drv' failed to build
```

Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/generated.nix:2563]: I tried to reproduce the failure but wasn't able to...
```
trying https://github.com/derekstride/tree-sitter-sql/archive/b8175006d9c8120d41cf40a4ef3711bbbbc08973.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  806k  100  806k    0     0   647k      0  0:00:01  0:00:01 --:--:-- 1064k
evaling implicit 'postFetch' string hook
unpacking source archive /build/b8175006d9c8120d41cf40a4ef3711bbbbc08973.tar.gz
calling 'unpackCmd' function hook '_defaultUnpack' /build/b8175006d9c8120d41cf40a4ef3711bbbbc08973.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/5fksaff7shpb8qwxa01mf43ybivvdbww-source.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-idQB8Wqw7lvU192y7+UgFvcwlmY71/mu9jJ4hRc4ud4=
```

Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/generated.nix:2563]: @vtuan10, have you been able to reproduce this ?

Comment (kalekseev) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/generated.nix:2563]: I can reproduce this with
```
❯ nix-shell -p nix-prefetch-git jq --run "nix hash to-sri sha256:\$(nix-prefetch-git --url https://github.com/derekstride/tree-sitter-sql --quiet --rev b8175006d9c8120d41cf40a4ef3711bbbbc08973 | jq -r '.sha256')"
sha256-ZURUEd7TokBIVVejRoBiXMQ1XwUEgMnzOhQiV+Tdpk0=
```

Comment (vtuan10) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/generated.nix:2563]: > @vtuan10, have you been able to reproduce this ?

Yes, I encountered this error as well when I updated my NixOS config. It also seems like there exists a PR already which fixes this issue: https://github.com/NixOS/nixpkgs/pull/332824

**Pull request merged**